cmd/demo/node.go

```diff
@@ -347,7 +363,11 @@ func (n *node) Close(args []string) error {
 		return errors.WithMessage(err, "sending final state for state closing")
 	}
 
-	return n.settle(peer)
+	if err := n.settle(peer); err != nil {
+		return errors.WithMessage(err, "settling")
+	}
+	fmt.Printf("\r🏁 Settled channel with %s.\n", peer.alias)
```

Comment:
No, `r` is not MacOS specific. This character tells the printer (e.g., the terminal) to move the cursor to the beginning of the line (see [Carriage return](https://en.wikipedia.org/wiki/Carriage_return)). I use it for overwriting the prefix `>` when printing.

I don't use `PrintfAsync` because `Close` is not triggered by an async call (e.g., a channel event). However, even though `Close` is not called asynchronously, the settle call contains an async print, which adds `>` to the next line. Hence, the `\r` is still necessary to remove.

This whole trouble comes from the fact that we cannot tell the go-prompt to clean print on an async event. To have a clean looking console without intermittent `>` prefix characters, we therefore use `PrintfAsync`, which first overwrites the existing `>`, then prints the message, and then adds the `>` again.

Reply:
I see, makes sense.